python/mxnet/array_function_protocol.py

```diff
@@ -0,0 +1,98 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Utils for registering NumPy array function protocol for mxnet.numpy ops."""
+
+import numpy as onp
+from . import numpy as mx_np  # pylint: disable=pylint=reimported
+from .numpy.multiarray import _NUMPY_ARRAY_FUNCTION_DICT
+
+
+def _implements(numpy_function):
+    """Register an __array_function__ implementation for MyArray objects."""
+    def decorator(func):
+        _NUMPY_ARRAY_FUNCTION_DICT[numpy_function] = func
+        return func
+    return decorator
+
+
+_NUMPY_ARRAY_FUNCTION_LIST = [
```

Comment:
is this the list of operators where our operator definition complies with the official numpy behavior?

Reply:
Yes, it is the operators in our codebase that is dispatchable through the array function protocol. Others will be dispatched through either fluent methods or array ufunc protocol, which will be implemented later.

Reply:
Awesome. We can use this list to track the full-compatibility with numpy operators once we start measuring it.